astropy/nddata/decorators.py

```diff
@@ -175,16 +187,20 @@ def wrapper(data, *args, **kwargs):
                         # If it's in the func_args it's trivial but if it was
                         # in the func_kwargs we need to compare it to the
                         # default.
-                        # FIXME: This obviously fails if the explicitly given
-                        # value is identical to the default. No idea how that
-                        # could be achieved without replacing the signature or
-                        # parts of it. That would have other drawbacks like
-                        # making double-decorating nearly impossible and
-                        # violating the sphinx-documentation for that function.
+                        # Comparison to the default is done by comparing their
+                        # identity, this works because defaults in function
+                        # signatures are only created once and always reference
+                        # the same item.
+                        # FIXME: Python interns some values, for example the
+                        # integers from -5 to 255 (any maybe some other types
+                        # as well). In that case the default is
+                        # indistinguishable from an explicitly passed kwarg
+                        # and it won't notice that and use the attribute of the
+                        # NDData.
```

Comment:
I think in practice this is not an issue because none of the supported properties can have a value like that. The exception *might* be units, though I'm having a little trouble imagining a real use case in which the data already has a unit, the function being called has a unit argument with a default unit, and the user tries to override that by specifying the unit rather than using the unit attached to the data.